### PR TITLE
Add all core testing infrastructure

### PR DESCRIPTION
**Pull request by taywrobel**
Adds all core automation and some testing harnesses.

### CODE REVIEW

**Comment by dgreenhalgh** [comment]
I'll read through all of this later, but I love the sound of it

**Comment by dgreenhalgh** [comment]
Could you add a setup description for the dependencies in our docs folder?

File: SoundStream-test/src/com/lastcrusade/soundstream/ete/harnesses/ConnectHarness.java [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+		solo.clickOnView(solo.getView(R.id.create_network_id));
+		assertVisible(false); // Should navigate away from fragment
+	}
+
+	public void pressJoinButton() {
+		solo.clickOnView(solo.getView(R.id.join_network_id));
+	}
+
+	public void answerBluetoothDialog(boolean respondYes) {
+		// FIXME: Doesn't work
```
Comment (reidbaker) [inline SoundStream-test/src/com/lastcrusade/soundstream/ete/harnesses/ConnectHarness.java:61]: Is this fix me supposed to happen before it gets pulled in? If not is there a ticket to fix this?

File: SoundStream-test/src/com/lastcrusade/soundstream/ete/harnesses/ConnectHarness.java [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+	public void joinNetwork() {
+		pressJoinButton();
+		answerBluetoothDialog(true);
+	}
+
+	public void assertBluetoothDialogShowing(final boolean expected) {
+		Condition con = new Condition() {
+			@Override
+			public boolean isSatisfied() {
+				// FIXME: Doesn't work
```
Comment (reidbaker) [inline SoundStream-test/src/com/lastcrusade/soundstream/ete/harnesses/ConnectHarness.java:75]: same as above

File: SoundStream-test/src/com/lastcrusade/soundstream/ete/harnesses/SoundStreamHarness.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+	public void assertMenuButtonActive(boolean expected) {
+		// FIXME: Doesn't work
+		View menuButton = solo.getView(android.R.id.home);
+		assertEquals("Menu not correctly enabled/disabled", expected,
+				menuButton.isEnabled());
+	}
+
+	public void assertTitleEquals(String expectedTitle) {
+		// FIXME: Doesn't grab title as it should. Should pull from UI directly.
```
Comment (reidbaker) [inline SoundStream-test/src/com/lastcrusade/soundstream/ete/harnesses/SoundStreamHarness.java:70]: same here

File: SoundStream-test/src/com/lastcrusade/soundstream/ete/suites/AllEteTests.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+/**
+ * A collection of all end to end test written for SoundStream. This should
+ * serve as the main class run through JUnit, as it contains the suite of all
+ * other test classes.
+ * 
+ * @author Taylor Wrobel
+ * 
+ */
+public class AllEteTests extends TestCase {
```
Comment (reidbaker) [inline SoundStream-test/src/com/lastcrusade/soundstream/ete/suites/AllEteTests.java:36]: Looks funny... is ete an abbreviation? Should we change this to AllETETests?

File: SoundStream-test/src/com/lastcrusade/soundstream/util/EteUtils.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+	public static boolean isFragmentVisible(Solo solo, Class<? extends Fragment> fragment) throws IllegalStateException{
+		if(!(solo.getCurrentActivity() instanceof CoreActivity)){
+			throw new IllegalStateException("Current Activity isn't CoreActivity, as expected");
+		}
+		
+		CoreActivity core = (CoreActivity)solo.getCurrentActivity();
+		
+		boolean vis = false;
+		
+		for(Fragment f : core.getAttachedFragments()){
```
Comment (reidbaker) [inline SoundStream-test/src/com/lastcrusade/soundstream/util/EteUtils.java:56]: I have no idea what this block of code does. Could we get some comments into this :poop:

File: SoundStream/res/layout/fragment_connect.xml [hunk 6]
```diff
@@ -61,6 +63,7 @@
         
         <ImageView
             android:contentDescription="@string/join_network"
+	        android:id="@+id/join_Img"
```
Comment (reidbaker) [inline SoundStream/res/layout/fragment_connect.xml:66]: Spacing looks funny here

**Comment by reidbaker** [comment]
I have a few comment but otherwise this looks great

File: SoundStream-test/src/com/lastcrusade/soundstream/ete/suites/AllEteTests.java (hunk 4 above)
Comment (taywrobel) [inline SoundStream-test/src/com/lastcrusade/soundstream/ete/suites/AllEteTests.java:36]: I would go with no.  This was a huge debate at work for a while, but we decided to only camel case the starts of acronyms.  In this case Ete is End-to-end.

Comment (dgreenhalgh) [inline SoundStream-test/src/com/lastcrusade/soundstream/ete/suites/AllEteTests.java:36]: I've seen it go both ways - for example I've seen the shorthand for identification as "ID" and "Id"

**Pull request merged**